app-backend/database/src/main/scala/com/azavea/rf/database/tables/Scenes.scala

```diff
@@ -401,11 +403,28 @@ object Scenes extends TableQuery(tag => new Scenes(tag)) with LazyLogging {
     * @param user    Results will be limited to user's organization
     */
   def deleteScene(sceneId: UUID, user: User)(implicit database: DB): Future[Int] = {
-    database.db.run {
+    val sceneQuery =
       Scenes
         .filterToSharedOrganizationIfNotInRoot(user)
         .filter(_.id === sceneId)
-        .delete
+    database.db.run {
+      sceneQuery.result
+    } flatMap { (scenes) =>
+      scenes.headOption match {
+        case Some(scene) =>
+          database.db.run {
+            sceneQuery
+              .filterToOwnerIfNotInRootOrganization(user)
+              .delete
+          } map { numDeleted =>
+            numDeleted match {
+              case 0 =>
+                throw IllegalRequestException(StatusCodes.ClientError(403)("Forbidden", "Error deleting scene: not authorized to modify this scene"))
```

Comment:
According to the postgres docs:
```
On successful completion, a DELETE command returns a command tag of the form

DELETE count
The count is the number of rows deleted. Note that the number may be less than the number of rows that matched the condition when deletes were suppressed by a BEFORE DELETE trigger. If count is 0, no rows were deleted by the query (this is not considered an error).
```
Since we don't plan on using database triggers anytime soon, I don't think this will be a problem.